src/compat/db_vdb.c

```diff
@@ -658,37 +666,47 @@ db_compile_statement_local (DB_SESSION * session)
 	}
     }
 
-  /* translate views or virtual classes into base classes */
-  statement_result = mq_translate (parser, statement);
-  if (!statement_result || pt_has_error (parser))
+  if ((statement->node_type == PT_INSERT && statement->info.insert.spec->info.spec.remote_server_name)
+      || (statement->node_type == PT_DELETE && statement->info.delete_.spec->info.spec.remote_server_name)
+      || (statement->node_type == PT_UPDATE && statement->info.update.spec->info.spec.remote_server_name)
+      || (statement->node_type == PT_MERGE && statement->info.merge.into->info.spec.remote_server_name))
```

Comment:
How about 'PT_IS_DML_DBLINK_QUERY' macro function?

Reply:
Ok, I modified it.